framework/wazuh/core/stats.py

```diff
@@ -219,59 +219,172 @@ def get_daemons_stats_(filename: str) -> list:
+    agent_id : Union[str, int]
+        The ID of the agent, which can be either a string or an integer.
+    daemon : str
+        The name of the daemon.
+
+    Returns
+    -------
+    bool
+        True if the daemon exists for the agent, False otherwise.
+    """
+    # Some daemons do not exist in agent 000
+    return not (is_agent_a_manager(agent_id) and daemon in {'agent'})
+
+
+def send_command_to_socket(dest_socket: str, command: str) -> dict:
+    """Send a command to a socket
+    Parameters
+    ----------
+    dest_socket : str
+        The destination socket path.
+    command : str
+        The command to send to the socket
+
+    Returns
+    -------
+    dict
+        Response from the socket
+
+    """
     try:
         s = wazuh_socket.WazuhSocket(dest_socket)
-    except Exception:
+    except WazuhException:
+        # Error connecting to socket
         raise WazuhInternalError(1121)
 
-    # Send message
-    s.send(command.encode())
-
-    # Receive response
     try:
-        rec_msg = s.receive().decode()
-    except ValueError:
-        raise WazuhInternalError(1118, extra_message="Data could not be received")
+        s.send(command.encode())
+        try:
+            return s.receive().decode()
+        except ValueError:
+            raise WazuhInternalError(1118, extra_message="Data could not be received")
+    finally:
+        s.close()
 
-    s.close()
 
-    # Format response
-    try:
-        data = json.loads(rec_msg)['data']
-        data.update((k, utils.get_utc_strptime(data[k], "%Y-%m-%d %H:%M:%S").strftime(common.DATE_FORMAT))
-                    for k, v in data.items() if k in {'last_keepalive', 'last_ack'})
-        return data
-    except Exception:
-        rec_msg = rec_msg.split(" ", 1)[1]
+def get_daemons_stats_from_socket(agent_id: str, daemon: str) -> dict:
+    """Get a daemon stats from an agent or manager.
+
+    Parameters
+    ----------
+    agent_id : str
+        Id of the agent to get stats from.
+    daemon : str
+        Name of the service to get stats from.
+
+    Returns
+    -------
+    dict
+        Object with daemon's stats.
+    """
+    if not agent_id or not daemon:
+        raise WazuhError(1307)
+
+    if not check_if_daemon_exists_in_agent(agent_id=agent_id, daemon=daemon):
+        raise WazuhError(1310)
+
+    dest_socket = get_stats_socket_path(agent_id=agent_id, daemon=daemon)
+    command = create_stats_command(agent_id=agent_id, daemon=daemon)
+
+    socket_msg = send_command_to_socket(dest_socket, command)
+    socket_response = json.loads(socket_msg)
+
+    # Handle error
+    if socket_response.get('error', 0) != 0:
+        rec_msg = socket_response.get('message', "")
         raise WazuhError(1117, extra_message=rec_msg)
+
+    data = socket_response['data']
+    data.update((k, utils.get_utc_strptime(data[k], "%Y-%m-%d %H:%M:%S").strftime(common.DATE_FORMAT))
+                for k, v in data.items() if k in {'last_keepalive', 'last_ack'})
```

Comment:
Something similar is happening here. Data received looks like:
```JSON
{
  "data": {
    "affected_items": [
      {
        "global": {
          "start": "2023-11-23 10:22:46",
          "end": "2023-11-23 10:51:46",
          "files": [
            {
              "location": "last -n 20",
              "events": 5,
              "bytes": 330,
              "targets": [
                {
                  "name": "agent",
                  "drops": 0
                }
              ]
            },
     ...
```

But the `if` condition inside the loop checks if the `global` key is `last_keepalive` or `last_ack`. Neither `last_keepalive` nor `last_ack` exist in the response. 

Also, at the `global` level there are no date-shaped keys, they are nested in one more key.